### PR TITLE
Simplify access to the SemanticModel in analyzers

### PR DESCRIPTION
**Pull request by meziantou**

### CODE REVIEW

File: src/Spectre.Console.Analyzer/Analyzers/NoConcurrentLiveRenderablesAnalyzer.cs [hunk 1]
```diff
@@ -38,9 +38,7 @@ protected override void AnalyzeCompilation(CompilationStartAnalysisContext compi
                     return;
                 }
 
-#pragma warning disable RS1030 // Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
-                var model = context.Compilation.GetSemanticModel(context.Operation.Syntax.SyntaxTree);
-#pragma warning restore RS1030 // Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
+                var model = context.Operation.SemanticModel!;
```
Comment (meziantou) [inline src/Spectre.Console.Analyzer/Analyzers/NoConcurrentLiveRenderablesAnalyzer.cs:41]: According to the doc it is safe to consider the SemanticModel to be non-null

[documentation](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.ioperation.semanticmodel?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk%28Microsoft.CodeAnalysis.IOperation.SemanticModel%29%3Bk%28DevLang-csharp%29%26rd%3Dtrue&view=roslyn-dotnet-4.3.0&WT.mc_id=DT-MVP-5003978)

> Optional semantic model that was used to generate this operation. Non-null for operations generated from source with [GetOperation(SyntaxNode, CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.semanticmodel.getoperation?view=roslyn-dotnet-4.3.0#microsoft-codeanalysis-semanticmodel-getoperation(microsoft-codeanalysis-syntaxnode-system-threading-cancellationtoken)) API and operation callbacks made to analyzers. Null for operations inside a [ControlFlowGraph](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.flowanalysis.controlflowgraph?view=roslyn-dotnet-4.3.0).

**Comment by patriksvensson** [comment]
@phil-scott-78 Do you have time to look at this?

**Comment by phil-scott-78** [comment]
If @meziantou submitted it, then I'll probably learn something cool - so, of course! I'll carve out time tonight

**Comment by meziantou** [comment]
> If @meziantou submitted it, then I'll probably learn something cool - so, of course! I'll carve out time tonight

I don't think there is a lot to learn from this PR, but I hope you'll like the next one https://github.com/spectreconsole/spectre.console/pull/1169 😉

**Comment by meziantou** [comment]
@phil-scott-78 Have you got time to review this PR? (and [others small PRs I've opened](https://github.com/spectreconsole/spectre.console/pulls/meziantou))

**Comment by phil-scott-78** [comment]
absolutely. thanks for the nudge, would have missed them otherwise.

**Review by phil-scott-78 — APPROVED** [review]

**Comment by phil-scott-78** [comment]
I think we got them all! Thanks @meziantou!

**Comment by meziantou** [comment]
That was fast 🚀
Thanks!

**Pull request merged**